gradle/spotless.gradle

```diff
@@ -17,6 +17,10 @@ spotless {
 		java {
 			ratchetFrom 'origin/main'
 			bumpThisNumberIfACustomStepChanges(1)
+			// `setMutators` will discard default mutators
+			// https://sonarsource.atlassian.net/browse/RSPEC-6212
+			cleanthat().version("2.9").sourceCompatibility(project.sourceCompatibility.toString()).clearMutators()
```

Comment:
Maybe the `sourceCompatibility` could be set here?

https://github.com/diffplug/spotless/blob/6789a1dc76d796aca31a1473c583066ad6400321/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java#L376-L378

Something like `if (isCleanThatStep(step)) && sourceCompatibility == null) { ...`

Reply:
I did not suceed doing so in the initial iteration. having a new look with your suggestion, I'm still stuck as I can not find `sourceCompatiblity` in `Project`: `https://github.com/gradle/gradle/blob/v7.6.1/subprojects/core-api/src/main/java/org/gradle/api/Project.java`, there is no `sourceCompatiblity`.

Yet, `project.sourceCompatibility` is OK in gradle files. What am I missing here?

Reply:
https://stackoverflow.com/a/72040924